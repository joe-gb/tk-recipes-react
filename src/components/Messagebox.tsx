import React from "react";
import styled from "styled-components";

export const Messagebox = styled.div`
  & {
    background: #ffffff;
    border-radius: 4px;
    border: 1px solid #fff;
    border-left: 4px solid #144fcc;
    padding: 0px 20px;

    & h3 {
      margin: 20px auto;
      color: #003097;
    }

    & p {
      margin: 20px auto;
      font-size: 14px;
      font-weight: 400;
      color: #1d2c3c;
    }
  }
`;

export const Successbox = styled(Messagebox)`
  & {
    background: #e5fff4;
    border: 1px solid #e5fff4;
    border-radius: 4px;
    border-left: 0px;
    & h3 {
      font-size: 16px;
      font-weight: 600;
      color: #000000cc;
    }
  }
`;

export const Errorbox = styled(Successbox)`
  & {
    background: #ffe5e5;
    border: 1px solid #ffe5e5;
  }
`;
