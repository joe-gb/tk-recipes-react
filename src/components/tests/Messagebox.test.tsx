import React from "react";
import { render } from "@testing-library/react";
import { Errorbox, Messagebox, Successbox } from "../Messagebox";

test("Messagebox snapshot test", () => {
  let { container } = render(<Messagebox>This is a message</Messagebox>);
  expect(container).toMatchSnapshot();
  expect(container.textContent).toBe("This is a message");
});

test("Successbox snapshot test", () => {
  let { container } = render(
    <Successbox>This is a success message</Successbox>
  );
  expect(container).toMatchSnapshot();
  expect(container.textContent).toBe("This is a success message");
});

test("Errorbox snapshot test", () => {
  let { container } = render(<Errorbox>This is an error message</Errorbox>);
  expect(container).toMatchSnapshot();
  expect(container.textContent).toBe("This is an error message");
});
