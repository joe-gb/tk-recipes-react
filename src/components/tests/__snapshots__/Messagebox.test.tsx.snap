// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Errorbox snapshot test 1`] = `
.c0 {
  background: #ffffff;
  border-radius: 4px;
  border: 1px solid #fff;
  border-left: 4px solid #144fcc;
  padding: 0px 20px;
}

.c0 h3 {
  margin: 20px auto;
  color: #003097;
}

.c0 p {
  margin: 20px auto;
  font-size: 14px;
  font-weight: 400;
  color: #1d2c3c;
}

.c1 {
  background: #e5fff4;
  border: 1px solid #e5fff4;
  border-radius: 4px;
  border-left: 0px;
}

.c1 h3 {
  font-size: 16px;
  font-weight: 600;
  color: #000000cc;
}

.c2 {
  background: #ffe5e5;
  border: 1px solid #ffe5e5;
}

<div>
  <div
    class="c0 c1 c2"
  >
    This is an error message
  </div>
</div>
`;

exports[`Messagebox snapshot test 1`] = `
.c0 {
  background: #ffffff;
  border-radius: 4px;
  border: 1px solid #fff;
  border-left: 4px solid #144fcc;
  padding: 0px 20px;
}

.c0 h3 {
  margin: 20px auto;
  color: #003097;
}

.c0 p {
  margin: 20px auto;
  font-size: 14px;
  font-weight: 400;
  color: #1d2c3c;
}

<div>
  <div
    class="c0"
  >
    This is a message
  </div>
</div>
`;

exports[`Successbox snapshot test 1`] = `
.c0 {
  background: #ffffff;
  border-radius: 4px;
  border: 1px solid #fff;
  border-left: 4px solid #144fcc;
  padding: 0px 20px;
}

.c0 h3 {
  margin: 20px auto;
  color: #003097;
}

.c0 p {
  margin: 20px auto;
  font-size: 14px;
  font-weight: 400;
  color: #1d2c3c;
}

.c1 {
  background: #e5fff4;
  border: 1px solid #e5fff4;
  border-radius: 4px;
  border-left: 0px;
}

.c1 h3 {
  font-size: 16px;
  font-weight: 600;
  color: #000000cc;
}

<div>
  <div
    class="c0 c1"
  >
    This is a success message
  </div>
</div>
`;
